packages/docs-site/docs/ref/domain/syntax.mdx

```diff
@@ -5,4 +5,142 @@ pagination_next: null
+
+Notably, the _domain_ schema are not _instructions_ of how to draw the _Set_ objects and relations between them. Such instructions of the specific diagram elements are provided by the _style_ schema, not the _domain_ schema. This allows multiple visual representations to be applied to objects from the same domain.
+
+There are four types of statements that can appear in the _domain_ schema.
+
+## Type Declarations
+
+A _type_ is a class of objects that a diagram works with. In the set-theory example, _Set_ is a type. The syntax for a type declaration is as follows:
+
+```
+type typename
+```
+
+This statement declares a type with name `typename`. After declaration, a type can be referred to in other parts of the _domain_ schema, as well as the _style_ and _substance_ schemas.
+
+For example, statement `type Set` declares a type with name `Set`. We can then use the name `Set` to later refer to this type.
+
+### Subtypes
+
+Conceptually, a type might be a _subtype_ of another type. As an example, in a chemistry domain, types _Hydrogen_ and _Oxygen_ might both be subtypes of the type _Atom_. _Penrose_ allows us to represent subtyping relations as follows:
+
+```
+type subtype_name <: supertype_name
+```
+
+Take the example of types _Hydrogen_, _Oxygen_, and _Atom_:
+
+```
+type Atom
+type Hydrogen <: Atom
+type Oxygen <: Atom
+```
+
+Subtyping describes "is-a" relationships between types: a Hydrogen "is-a[n]" Atom; an Oxygen "is-a[n]" Atom. Using the "is-a" interpretation, since an Atom "is-a[n]" Atom, inherently, for all type `A`, we have that `A` is a subtype of `A`. This interpretation of subtyping is consistent with "inheritance" in many object-oriented programming languages.
+
+_Penrose_ adopts this interpretation: if _Penrose_ expects to see an object with type `A`, then it will also accept an object of type `B`, as long as `B` is a subtype of `A`. We say that "`B` **matches** `A`."
+
+## Predicate Declarations
+
+A _predicate_ represents mathematical or logical statements regarding zero or more objects or other statements. To declare a predicate, we write
+
+```
+predicate predicate_name (argument_list)
+```
+
+where
+
+- `predicate_name` is the name of the predicate, which can be referred to in the _substance_ and _style_ schemas; and
+- `argument_list` describes the types of objects that this predicate works with. We will go more into its details below.
+
+The arguments in `argument_list` contains the types of objects that this predicate accepts, separated by commas. Here is a _domain_ schema that contains a couple of type declarations and four different, valid, predicate declarations:
+
+```
+type FirstType, SecondType
+
+predicate P1 (FirstType, SecondType)
+predicate P2 (SecondType a1, SecondType a2)
+predicate P3 ()
+predicate P4 (FirstType, SecondType, SecondType)
```

Comment:
Speaking of consistency: it looks like in some places we have a space between the predicate name and the open parenthesis for the argument list, and in other places we don't. Can we choose one and stick to that? My vote is for no space.

Reply:
I feel like adding spaces allows readers to better understand the structure of the exprs

Reply:
OK! Yeah, I don't have much of a preference between the two choices; I just want to be consistent. @joshsunshine @wodeni thoughts?

Reply:
Concurring with @samestep here on the formatting.

Reply:
I can make it to always have spaces in the docs